### DISCUSSION TITLE
"Rerun" fails to parse when using Reusable workflows

### DISCUSSION THREAD

**Discussion by connorjclark** [community/community#40777] [Actions]
### Select Topic Area

Bug

### Body

I have some reusable workflows (nested, actually) that will fail to parse on a re-run.

The original run: https://github.com/connorjclark/ZeldaClassic/actions/runs/3614685421/jobs/6091319667
The failed re-run: https://github.com/connorjclark/ZeldaClassic/actions/runs/3614685421
The error:
```
error parsing called workflow "./.github/workflows/test.yml": workflow was not found. See https://docs.github.com/actions/learn-github-actions/reusing-workflows#access-to-reusable-workflows for more information.
```

The docs say:

> Re-running failed jobs or a specific job in a workflow will use the reusable workflow from the same commit SHA of the first attempt. For more information about re-running failed jobs in a workflow, see ["Re-running failed jobs in a workflow"](https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs#re-running-failed-jobs-in-a-workflow). For more information about re-running a specific job in a workflow, see ["Re-running a specific job in a workflow"](https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs#re-running-a-specific-job-in-a-workflow).

So I'd expect the action to re-run correctly using the current commit.

________

One possible fix is to use the public syntax for the workflow (`ArmageddonGames/ZQuestClassic/.github/workflows/build.yml@main`), but this solution _does not work_ for me because my project expects development (including modifying workflows) to work in forks, and there doesn't seem to be a way to use any interpolation (to reference the current repo context) in `uses:`.

**Comment by ossanna16** [comment]
Hi there @connorjclark and welcome to our community! Thank you for reporting this!

**Reply by connorjclark** [reply]
HI Anna - is this the right place to report issues like this? Is there a better way to submit bugs for GitHub Actions?

**Comment by connorjclark** [comment]
This remains an issue. Another failed retry: https://github.com/connorjclark/ZeldaClassic/actions/runs/5101682005

Note that "rerun all" works. Just not "rerun failures".

**Comment by psychobolt** [comment]
Which case ends up in a bad state of empty job list for the attempt to re-run all jobs. 

![failure](https://github.com/community/community/assets/560721/bce09468-0a90-4910-bd0d-8a3140e4f1b6)


Trying to rerun a list of empty jobs obviously ends up with another failure: https://github.com/psychobolt/vite-storybook-boilerplate/actions/runs/7367008714 , surprisingly a return with the parse error once again...
